source/common/json/config_schemas.cc

```diff
@@ -505,6 +561,26 @@ const std::string Json::Schema::TCP_PROXY_NETWORK_FILTER_SCHEMA(R"EOF(
             }
           },
           "additionalProperties": false
+        },
+        "access_log" : {
+          "type" : "array",
+          "items" : {
+            "type" : "object",
+            "properties" : {
+              "path" : {"type" : "string"},
+              "format" : {"type" : "string"},
+              "filter" : {
+                "type" : "object",
+                "oneOf" : [
+                  {"$ref" : "#/definitions/duration"},
+                  {"$ref" : "#/definitions/runtime"},
+                  {"$ref" : "#/definitions/logical_filter"}
+                ]
+              }
+            },
+            "required" : ["path"],
+            "additionalProperties" : false
+          }
```

Comment:
one suggestion. Can we just do this in the v2 api protos, so that it starts becoming the one place where we add stuff to configs? I just synced the whole thing and I am hoping to not diverge anymore.

We could just copy the access log stuff in filter/http/ into filter/ or something along those lines.

Reply:
Right now, tcp_proxy is configured via json, not via v2 APIs.  Until that is converted, this is the only way to do it, I think.

Reply:
That sucks! Atleast could you send a pr to data plane api to add the corresponding configuration bits? It’s fine if it’s duplicated for the moment

Reply:
Yes, I'll do that.  I wanted to get buy-in that this is the right approach.  Once I have that, it should be a one-line change in the api (adding one config option to tcp_proxy, which uses the existing http-access-log type/message).

Reply:
FWIW, making TCP proxy configurable via v2 is probably a relatively easy PR, comparable to https://github.com/envoyproxy/envoy/pull/1952. The proto in `data-plane-api` is already there.